File: benchmarks/mixed_types.py
n = 30000
int_total = 0
float_total = 0.0
str_len_total = 0
list_total = 0
cache = {}
i = 0
while i < n:
    int_total = int_total + i
    float_total = float_total + i * 1.5
    s = f"n{i}"
    str_len_total = str_len_total + len(s)
    xs = [i, i + 1]
    list_total = list_total + xs[0] + xs[1]
    k = i % 100
    if k in cache:
        cache[k] = cache[k] + 1
    else:
        cache[k] = 1
    i = i + 1
cache_total = 0
j = 0
while j < 100:
    cache_total = cache_total + cache[j]
    j = j + 1
print(int_total)
print(float_total)
print(str_len_total)
print(list_total)
print(cache_total)

File: benchmarks/cpp/mixed_types.cpp
#include <cstdio>
#include <string>
#include <unordered_map>

int main() {
    const int n = 30000;
    long long int_total = 0;
    double float_total = 0.0;
    long long str_len_total = 0;
    long long list_total = 0;
    std::unordered_map<int, long long> cache;
    for (int i = 0; i < n; i++) {
        int_total += i;
        float_total += i * 1.5;
        std::string s = "n" + std::to_string(i);
        str_len_total += (long long)s.size();
        long long xs[2] = {i, i + 1};
        list_total += xs[0] + xs[1];
        int k = i % 100;
        auto it = cache.find(k);
        if (it != cache.end()) it->second += 1;
        else cache[k] = 1;
    }
    long long cache_total = 0;
    for (int j = 0; j < 100; j++) cache_total += cache[j];
    printf("%lld\n", int_total);
    printf("%f\n", float_total);
    printf("%lld\n", str_len_total);
    printf("%lld\n", list_total);
    printf("%lld\n", cache_total);
}
